### ISSUE TITLE
link-hint-open-link uses wrong browser

### ISSUE THREAD

**Issue by simendsjo** [syl20bnr/spacemacs#11874]
I'm stumped here... `link-hint-open-link` uses `firefox` if it's installed even though I'm using chromium as my default browser.

The following actions all opens chromium
- xdg-open "someurl"
- (browse-url "someurl")
- (browse-url-generic "someurl")
- (link-hint-open-link) ;; when firefox is *not* installed

but when firefox is installed, (link-hint-open-link) opens firefox. I've tried grepping my entire package repo without finding out why it does this... Any tips?

**Comment by bsless** [comment]
Hi @simendsjo, can you please check what's the value of `browse-url-browser-function`?
You can use
`SPC h d v browse-url-browser-function`
I'm guessing it's using some default value.
If it's indeed a default value different than Chromium, you can add this piece of code to your `user-init`:

```elisp
(setq browse-url-browser-function 'browse-url-generic
          browse-url-generic-program "chromium-browser")
```

**Comment by simendsjo** [comment]
Ok, this is rather strange, as it works now. Unless someone fixed something very recently (I've upgraded the packages and OS), there's something very strange going on in my system.. I reinstalled firefox, and it still opened chromium. Restarted emacs, still opened chromium. Restarted the OS, still opened chromium.

My settings is unchanged `browse-url-browser-function` is `‘browse-url-generic`, `browse-url-generic` is `"xdg-open"`, and no mime types is changed.

I really have no idea what might have happened here, but I suspect there's some odd case which just happens to be temporarily fixed for the time being... Maybe not related to link-hint at all..?
